plugins/extract/detect/s3fd_amd.py

```diff
@@ -0,0 +1,480 @@
+        return x
+
+
+class O2K_ReduceLayer(keras.engine.Layer):
+    def __init__(self, axes=None, keepdims=True, **kwargs):
+        self._axes = [axes] if isinstance(axes, int) else axes
+        self._keepdims = bool(keepdims)
+        super(O2K_ReduceLayer, self).__init__(**kwargs)
+
+    def get_config(self):
+        config = super(O2K_ReduceLayer, self).get_config()
+        config.update({
+            'axes': self._axes,
+            'keepdims': self._keepdims
+        })
+        return config
+
+    def compute_output_shape(self, input_shape):
+        if self._axes is None:
+            return (1,)*len(input_shape) if self._keepdims else tuple()
+        ret = list(input_shape)
+        for i in sorted(self._axes, reverse=True):
+            if self._keepdims:
+                ret[i] = 1
+            else:
+                ret.pop(i)
+        return tuple(ret)
+
+    def call(self, x, *args):
+        raise NotImplementedError()
+
+
+class O2K_Sum(O2K_ReduceLayer):
+    def call(self, x, *args):
+        return K.sum(x, self._axes, self._keepdims)
+
+
+class O2K_Sqrt(keras.engine.Layer):
+    def call(self, x, *args):
+        return K.sqrt(x)
+
+
+class O2K_Pow(keras.engine.Layer):
+    def call(self, x, *args):
+        return K.pow(*x)
+
+
+class O2K_ConstantLayer(keras.engine.Layer):
+    def __init__(self, constant_obj, dtype, **kwargs):
+        self._dtype = np.dtype(dtype).name
+        self._constant = np.array(constant_obj, dtype=self._dtype)
+        super(O2K_ConstantLayer, self).__init__(**kwargs)
+
+    def call(self, *args):
+        data = K.constant(self._constant, dtype=self._dtype)
+        return data
+
+    def compute_output_shape(self, input_shape):
+        return self._constant.shape
+
+    def get_config(self):
+        config = super(O2K_ConstantLayer, self).get_config()
+        config.update({
+            'constant_obj': self._constant,
+            'dtype': self._dtype
+        })
+        return config
+
+
+class O2K_Div(O2K_ElementwiseLayer):
+    def call(self, x, *args):
+        return x[0] / x[1]
+
+
+class S3fd_amd():
+    """ Keras Network """
+    def __init__(self, model_path, target_size, confidence):
+        logger.debug("Initializing: %s: (model_path: '%s')",
+                     self.__class__.__name__, model_path)
+        self.model_path = model_path
+        self.confidence = confidence
+        self.model = self.load_model()
+        logger.debug("Initialized: %s", self.__class__.__name__)
+
+    def load_model(self):
+        """ Load the keras Model and weights """
+        logger.verbose("Initializing S3FD_amd Network model...")
+        layers = {
+            'O2K_Add': O2K_Add, 'O2K_Slice': O2K_Slice,
+            'O2K_Sum': O2K_Sum, 'O2K_Sqrt': O2K_Sqrt,
+            'O2K_Pow': O2K_Pow, 'O2K_ConstantLayer': O2K_ConstantLayer,
+            'O2K_Div': O2K_Div
+        }
+        model = keras.models.load_model(self.model_path, custom_objects=layers)
+        return model
```

Comment:
This hangs on Tensorflow Backend. Unfortunately Keras does not play nicely with threading with the TF backend.

You can make it play nicely by adding the following (unfortunately protected access) function prior to returning the model:
```py
        model._make_predict_function()  # pylint: disable=protected-access
```